presto-spi/src/main/java/com/facebook/presto/spi/function/FunctionDependency.java

```diff
@@ -28,4 +31,14 @@
     String returnType();
 
     String[] argumentTypes();
+
+    Convention convention() default @Convention($notSpecified = true, arguments = {}, result = FAIL_ON_NULL);
+
+    public static @interface Convention {
```

Comment:
`public static` is the default. You can remove them.

In general, code gets messy if there are more than 1 way for the basic stuff. In Presto code base, in these situations, we always choose to elide.